lsp/src/tree_sitter.rs

```diff
@@ -57,50 +60,58 @@ fn create_attribute(node: Node<'_>, source: &str) -> Option<Position> {
     return None;
 }
 
-fn get_position_by_query(query: &str, node: Node<'_>, source: &str) -> Option<Position> {
+fn get_position_by_query(
+    query: &str,
+    node: Node<'_>,
+    source: &str,
+    trigger_point: Point,
+) -> Option<Position> {
+    // See: https://tree-sitter.github.io/tree-sitter/using-parsers#query-syntax
     let query = Query::new(tree_sitter_html::language(), query)
         .expect(&format!("get_position_by_query invalid query {query}"));
-    let mut cursor = QueryCursor::new();
-    let matches = cursor.matches(&query, node, source.as_bytes());
+    let mut cursor_qry = QueryCursor::new();
+    let matches = cursor_qry.matches(&query, node, source.as_bytes());
 
     debug!("get_position_by_query node {:?}", node.to_sexp());
 
-    // FIXME this only covers suggestion for attributes at the end of the tag!
-    let first_match = matches.last()?;
-
     let capture_names = query.capture_names();
-    debug!("get_position_by_query captures {:?}", capture_names);
-
-    debug!("get_position_by_query first_match {:?}", first_match);
-
-    let props = first_match
-        .captures
-        .iter()
-        .fold(HashMap::new(), |mut props, capture| {
-            let name = capture_names[capture.index as usize].to_owned();
-            let value = capture
-                .node
-                .utf8_text(source.as_bytes())
-                .expect(&format!("failed to parse capture value for '{name}'"))
-                .to_owned();
-
-            props.insert(name, value);
+    debug!("capture_names: {:?}", capture_names);
+
+    let props = matches
+        .into_iter()
+        .fold(HashMap::new(), |mut props, match_| {
+            match_
+                .captures
+                .iter()
+                .filter(|capture| capture.node.start_position() <= trigger_point)
```

Comment:
I changed a bit this approach but you are correct, because the node may end after the trigger point for instance 
`<div hx-target="|" ...` in the previous implementation it needed to understand what was else the node has or for the cases where there was a mismatch of quotes `<div hx-foo="|  class="bar">` the Errored node was at end, but I changed a bit and now it filters the nodes before the analysis happen